src/client/app/components/meters/MeterModalEditComponent.tsx

```diff
@@ -0,0 +1,567 @@
+/* eslint-disable @typescript-eslint/indent */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react';
+import { Modal, Button, Dropdown } from 'react-bootstrap';
+import { GPSPoint } from 'utils/calibration';
+import '../../styles/meter-edit-modal.css';
+
+interface meterTypes {
+  MAMAC?: string;
+  METASYS?: string;
+  OBVIUS?: string;
+  OTHER?: string
+}
+interface MetaDataProps {
+  onhide: () => void;
+  onSaveChanges: () => void;
+  handleIdentifierChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onCancel: () => void;
+  show: boolean;
+  id: number;
+  identifier: string;
+  name: string;
+  units?: string;
+  meterType?: string;
+  gps?: GPSPoint;
+  Area: number;
+  displayable: boolean;
+  enabled: boolean;
+  meterAddress?: string;
+  graphicUnit: number;
+  notes: string;
+  cumulative: boolean;
+  cumulativeReset: boolean;
+  cumulativeResetStart: string;
+  cumulativeResetEnd: string;
+  endOnlyTime: boolean;
+  readingGap: string;
+  readingVariation: number;
+  readingDuplication: number;
+  timesort: boolean;
+  startTimestamp: string;
+  endTimestamp: string;
```

Comment:
I grabbed all the values from the DB file that creates the meter object itself. I wasn't sure which one we would be using but since we were incorporating 22 new values I had just grabbed all of the values that the meter can keep track of.